src/briefcase/exceptions.py

```diff
@@ -137,11 +137,10 @@ def __init__(self, error_msg):
         super().__init__(msg=error_msg, skip_logfile=True)
 
 
-class TemplateUnsupportedVersion(BriefcaseCommandError):
-    def __init__(self, briefcase_version):
-        self.briefcase_version = briefcase_version
+class InvalidTemplateBranch(BriefcaseCommandError):
+    def __init__(self, template_repo, branch):
         super().__init__(
-            f"Could not find a template branch for Briefcase {briefcase_version}."
+            f"{branch} did not match any branches in template repository {template_repo}."
```

Comment:
The `!r` annotation is helpful for differentiating which part of text is user specified; it wasn't required previously because it was assumed the branch was always in the form of a Briefcase version number.
```suggestion
            f"Could not find a branch named {branch!r} in template repository {template_repo!r}."
```

Reply:
TIL, thanks!